src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -169,7 +169,24 @@ private static TemplateMatcher GetInstanceRaiseEventRoute()
             Stopwatch stopwatch = Stopwatch.StartNew();
             while (true)
             {
-                DurableOrchestrationStatus status = await client.GetStatusAsync(instanceId);
+                DurableOrchestrationStatus status = null;
+
+                if (client is DurableClient durableClient && durableClient.DurabilityProvider.SupportsPollFreeWait)

```

Comment:
The syntax of `client is DurableClient durableClient` confuses me a bit. I would have expected this to be simply `client is DurableClient`, which would be a type-check. What is this achieving?

Reply:
the name 'durableClient' gets bound and is used right after the &&.